tests/unit/pipeline-selector.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { createLocalVue, shallowMount } from '@vue/test-utils';
+import Vuex from 'vuex';
+
+import PipelineSelector from '@/components/PipelineSelector.vue';
+
+import { setupMockStore } from './utils';
+
+const localVue = createLocalVue();
+localVue.use(Vuex);
+
+describe('PipelineSelector', () => {
+  it('should instantiate', () => {
+    const wrapper = shallowMount(PipelineSelector, { store: setupMockStore(), localVue });
+    expect(wrapper.exists()).toBeTruthy();
+  });
+
+  it('should display all pipeline names in a select', () => {
+    const store = setupMockStore({
+      pipelines: {
+        pipeline1: [],
+        pipeline2: [],
+        pipeline3: [],
+      },
+    });
+    const wrapper = shallowMount(PipelineSelector, {
+      store,
+      localVue,
+    });
+    const select = wrapper.find('select');
+    expect(select.exists()).toBeTruthy();
+    const options = select.findAll('option');
+    expect(options).toHaveLength(3);
+    expect(options.at(0).text()).toEqual('pipeline1');
+    expect(options.at(1).text()).toEqual('pipeline2');
+    expect(options.at(2).text()).toEqual('pipeline3');
+  });
+
+  it('should display the selected pipeline if any', () => {
+    const store = setupMockStore({
+      currentPipelineName: 'pipeline2',
+      pipelines: {
+        pipeline1: [],
+        pipeline2: [],
+        pipeline3: [],
+      },
+    });
+    const wrapper = shallowMount(PipelineSelector, {
+      store,
+      localVue,
+    });
+    const select = wrapper.find('select');
+    expect((select.findAll('option').at(1).element as HTMLOptionElement).selected).toBeTruthy();
+  });
+
+  it('should update the currentPipelineName when a pipeline is selected', () => {
+    const store = setupMockStore({
+      currentPipelineName: 'pipeline2',
+      pipelines: {
+        pipeline1: [],
+        pipeline2: [],
+        pipeline3: [],
+      },
+    });
+    const wrapper = shallowMount(PipelineSelector, {
+      store,
+      localVue,
+    });
+    const select = wrapper.find('select');
+    select
+      .findAll('option')
+      .at(2)
+      .setSelected();
+    select.trigger('input');
+    expect(store.state.vqb.currentPipelineName).toStrictEqual('pipeline3');
```

Comment:
```suggestion
    expect(store.state.vqb.currentPipelineName).toEqual('pipeline3');
```

Reply:
Thanks for pointing that! I thought was mislead into a rough translation of chai's `eql` and `equal`. Turns out that jest does more than just checking for `==` and `===`.